pkg/controller/statspuller/statspull.go

```diff
@@ -16,23 +16,35 @@
 package statspuller
 
 import (
+	"github.com/google/exposure-notifications-verification-server/internal/clients"
 	"github.com/google/exposure-notifications-verification-server/pkg/config"
 	"github.com/google/exposure-notifications-verification-server/pkg/database"
 	"github.com/google/exposure-notifications-verification-server/pkg/render"
 )
 
 // Controller is a stats controller.
 type Controller struct {
-	config *config.StatsPullerConfig
-	db     *database.Database
-	h      render.Renderer
+	defaultKeyServerClient *clients.KeyServerClient
+	config                 *config.StatsPullerConfig
+	db                     *database.Database
+	h                      render.Renderer
 }
 
 // New creates a new stats-pull controller.
-func New(cfg *config.StatsPullerConfig, db *database.Database, h render.Renderer) *Controller {
-	return &Controller{
-		config: cfg,
-		db:     db,
-		h:      h,
+func New(cfg *config.StatsPullerConfig, db *database.Database, h render.Renderer) (*Controller, error) {
```

Comment:
I’d recommend passing the client into the controller instead of relying on the controller to build from configuration. It’s a nicer separation, esp for testing, and it removes the need to return an error